### PR TITLE
BFD-3795: Add resources to sonarqube pipeline job

### PR DESCRIPTION
**Pull request by mjburling**
**JIRA Ticket:**
BFD-3795

### What Does This PR Do?
- Adds additional resources to the container host for testcontainers.org to work correctly.
- Increases heapsize for the maven build process
- Removes a duplicate entry from the rda-grpc pom for jakarta persistence
- Clarifies the _nightly_ sonarqube jenkins pipeline execution

### What Should Reviewers Watch For?
Read through the solution and my own review comments to ensure that everything here makes sense.

### What Security Implications Does This PR Have?

Please indicate if this PR does any of the following:  

* Adds any new software dependencies
* Modifies any security controls
* Adds new transmission or storage of data
* Any other changes that could possibly affect security? 

* [x] I have considered the above security implications as it relates to this PR. (If one or more of the above apply, it cannot be merged without the ISSO or team security engineer's (`@sb-benohe`) approval.) 

### Validation

Have you fully verified and tested these changes? Is the acceptance criteria met? Please provide reproducible testing instructions, code snippets, or screenshots as applicable.

- unit/integration tests pass as part of CI
- **this** branch builds successfully in the jenkins sonarqube pipeline
-  similar changes run successfully against the master branch

### CODE REVIEW

File: apps/build.groovy [hunk 1]
```diff
@@ -134,10 +134,13 @@ EOF
 
     dir ('apps') {
         quietFlags = verboseMaven ? '' : '--quiet --batch-mode'
-        if (runTests) {
-            sh "mvn ${quietFlags} --threads 1C -Dmaven.javadoc.skip=true -Dcheckstyle.skip -Dmaven.build.cache.enabled=false -Dmaven.jacoco.skip=false clean install"
-        } else {
-            sh "mvn ${quietFlags} --threads 1C --update-snapshots -DskipITs -DskipTests -Dmaven.javadoc.skip=true -Dmaven.build.cache.enabled=false clean verify"
+        withEnv(['MAVEN_OPTS=-Xms1024m -Xmx1024m']) {
```
Comment (mjburling) [inline apps/build.groovy:137]: This modest doubling of the heap size from 512m to 1024m for maven isn't unreasonable, but it's largely unrelated to the current issues of heap exhaustion. Instead, the resource exhaustion _seems_ to be occurring in the supporting infrastructure for testcontainers.org. This won't hurt, though!

File: apps/bfd-pipeline/bfd-pipeline-rda-grpc/pom.xml [hunk 2]
```diff
@@ -205,11 +205,6 @@
             <artifactId>aws-crt</artifactId>
             <version>${aws.crt.version}</version>
         </dependency>
-        <dependency>
-            <groupId>jakarta.persistence</groupId>
-            <artifactId>jakarta.persistence-api</artifactId>
-            <version>3.2.0-M2</version>
-        </dependency>
```
Comment (mjburling) [inline apps/bfd-pipeline/bfd-pipeline-rda-grpc/pom.xml:212]: We ignore milestone versions for the purposes of the [maven-versions plugin](https://www.mojohaus.org/versions/versions-maven-plugin/index.html) under `apps/rules.xml`, however, that only applies when running the plugin explicitly, e.g. `mvn versions:display-dependency-updates`. This mismatch has been throwing errors which I only noticed with a verbose maven output when diagnosing the issues here:
> [WARNING] 'dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: jakarta.persistence:jakarta.persistence-api:jar -> version ${jakarta.persistence-api.version} vs 3.2.0-M2 @ line 208, column 21

Changes to `apps/bfd-pipeline/bfd-pipeline-rda-grpc/pom.xml` and `apps/pom.xml` are merely targets of opportunity here.

File: ops/jenkins/bfd-run-sonarqube/Jenkinsfile [hunk 3]
```diff
@@ -37,10 +37,10 @@ spec:
       mountPath: /var/run
     resources:
       requests:
-        memory: '8192Mi'
+        memory: '16384Mi'
         cpu: '4000m'
       limits:
-        memory: '8192Mi'
+        memory: '16384Mi'
```
Comment (mjburling) [inline ops/jenkins/bfd-run-sonarqube/Jenkinsfile:43]: This aligns the resources for the test container infrastructure with the[ sizing for hosted github actions runners](https://docs.github.com/en/actions/using-github-hosted-runners/using-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories) where resource exhaustion hasn't historically been a problem.

File: ops/jenkins/bfd-run-sonarqube/Jenkinsfile [hunk 4]
```diff
@@ -65,9 +65,13 @@ spec:
...
 
-    // Run Every day at 22:00 PM Eastern / 03:00 AM UTC
+
     triggers {
-        cron(env.BRANCH_NAME == 'master' ? '00 03 * * 1-5' : '')
+        // UTC-time dictates that we look at Tue-Sat morning instead of Mon-Fri evening
+        // https://www.worldtimebuddy.com/?qm=1&lid=100,5128581&h=100&date=2024-12-17&sln=3-4&hf=1
+        // https://crontab.guru/#00_03_*_*_2-6
+        // At 03:00 on every day-of-week from Tuesday through Saturday.
+        cron(env.BRANCH_NAME == 'master' ? '00 03 * * 2-6' : '')
```
Comment (mjburling) [inline ops/jenkins/bfd-run-sonarqube/Jenkinsfile:74]: Another target of opportunity. Jenkins runs in UTC. With the zero-offset, this was actually running Sunday evening through Thursday evening. Here, we run at 0300 UTC starting Tuesday (which is Monday evening in ET) through Saturday at 0300 (Friday evening in ET), making the historical Jenkins executions of this pipeline a little easier to follow as a result.

**Review by dondevun — DISMISSED** [review]

**Review by aschey-forpeople — APPROVED** [review]

**Review by malessi — DISMISSED** [review]

**Review by MahiFentaye — APPROVED** [review]

**Review by aschey-forpeople — APPROVED** [review]

**Review by dondevun — APPROVED** [review]

**Pull request merged**